yt_dlp/extractor/rokfin.py

```diff
@@ -0,0 +1,561 @@
+# coding: utf-8
+from json import dumps
+from math import ceil
+import itertools
+import datetime
+from re import finditer
+from urllib.parse import urlparse
+from .common import InfoExtractor, SearchInfoExtractor
+from ..utils import (
+    unified_timestamp,
+    try_get,
+    traverse_obj,
+    int_or_none,
+    bool_or_none,
+    float_or_none,
+    str_or_none,
+    url_or_none,
+    ExtractorError,
+)
+
+
+class RokfinIE(InfoExtractor):
+    _NETRC_MACHINE = 'rokfin'
+
+
+class RokfinSingleVideoIE(RokfinIE):
+    _META_DATA_BASE_URL = 'https://prod-api-v2.production.rokfin.com/api/v2/public/'
+    _RECOMMENDED_VIDEO_BASE_URL = 'https://rokfin.com/'
+    _CHANNEL_BASE_URL = 'https://rokfin.com/'
```

Comment:
I agree. The class hierarchy was different in earlier versions. The reason it looks this way now is that I am moving (slowly) towards site logins. I can change the class hierarchy back, if you insist. Otherwise, can you let that slide? It's temporary anyway.

Reply:
Why would implementing login require 3 separate base classes?

Reply:
RokfinIE is the root that contains the login logic. RokfinSingleVideoIE is the base class for posts and streams. RokfinPlaylistIE inherits from RokfinIE directly. Both RokfinStackIE and RokfinChannelIE inherit from RokfinPlaylistIE. All this is subject to change, of course.